### DISCUSSION TITLE
How to give runner filesystem permissions

### DISCUSSION THREAD

**Discussion by nmeri17** [paratestphp/paratest#813] [Q&A]
When I run the entire test suite, invocation fails when it reaches a particular folder containing tests that access the filesystem. The error message is thus:

> [28-Mar-2023 10:15:15 UTC] PHP Notice:  file_get_contents(): Read of 8192 bytes failed with errno=13 Permission denied in C:\wamp64\www\suphle\vendor\phpunit\phpunit\src\Runner\DefaultTestResultCache.php on line 109

Then one of the tests fails. If I run that test alone or the folder, everything passes. If I try with regular phpunit, everything passes.

Let me also use this opportunity to mention [this impediment](https://github.com/paratestphp/paratest/issues/576#issuecomment-1486058054) instead of opening a fresh issue. Perhaps, it'll aid me understand what's going on instead of working in the dark

**Comment by Slamdunk** [comment]
It is likely you have a race condition that only occurs within the concurrency operations of parallel tests running.

Which means that multiple tests use the same I/O device (i.e. a simple file on the filesystem) and they step on each other.

Avoid this by using unique file names for each TestCase, for example:
```php
protected function setUp(): void
{
    $this->file = __DIR__ . '/test_file_' . sha1(uniqid(__METHOD__));
}
```

**Comment by Slamdunk** [comment]
> how to output/unmute phpunit output while using paratest

You can't, sorry

**Comment by nmeri17** [comment]
Placing the assignment after `parent::setUp` worked.

Brother, please show me how to output/unmute phpunit output while using paratest

**Comment by nmeri17** [comment]
> You can't, sorry

That's very sad. The test only picks up 2 cores/processors, so the difference against phpunit is barely noticeable. Same with bumping it to 5 or whatever. In addition, I can't see what's going on. Only difference is lower memory consumption

Noob question: I thought phpunit runs were going to fail after adding the `$this->file` you suggested, but it surprisingly doesn't. How is it that dynamically setting test filename to a non-existent value causes them to "avoid stepping on each other [while trying to access the dummy file?]"? The test classes already have unique names, so what's special about the random values? Is the IO device created by my test and logging the name of the creator (such that it fails if actual creator matches the value of `$test->file`)?
